tests/credentials_provider_ecs_tests.c

```diff
@@ -536,6 +557,87 @@ static int s_credentials_provider_ecs_basic_success(struct aws_allocator *alloca
 
 AWS_TEST_CASE(credentials_provider_ecs_basic_success, s_credentials_provider_ecs_basic_success);
 
+static int s_credentials_provider_ecs_mocked_server_basic_ipv4_invalid(struct aws_allocator *allocator, void *ctx) {
```

Comment:
There's already a system for mocking HTTP calls in auth tests. Is there a really good reason for adding another?